snippets/fsharp/VS_Snippets_CLR_System/system.console.class/fs/example3.fs

```diff
@@ -0,0 +1,169 @@
+﻿
+// dump a range of Unicode characters as a 16x16 array
+// <Snippet4>
+open System
+open System.IO
+open System.Globalization
+open System.Text
+
+type uint = uint32
+
+let roundDownToMultipleOf (b:uint) (u:uint) : uint =
+    u - (u % b)
+
+let roundUpToMultipleOf (b:uint) (u:uint) : uint =
+    roundDownToMultipleOf b u |> (+) b
+
+let displayRange (start:uint) (``end``:uint) : unit =
+
+    let mutable upperRange : uint = 0x10FFFFu
+    let mutable surrogateStart : uint = 0xD800u
+    let mutable surrogateEnd : uint = 0xDFFFu
+   
+    let mutable start = start
+    let mutable ``end`` = ``end``
+    if ``end`` <= start
+    then
+        let t = start
+        start <- ``end``
+        ``end`` <- t
+
+    // Check whether the start or end range is outside of last plane.
+    if start > upperRange
+    then
+        raise (ArgumentException(String.Format("0x{0:X5} is outside the upper range of Unicode code points (0x{1:X5})", start, upperRange)))
```

Comment:
Prefer `invalidArg`:

```fsharp
invalidArg "start" (String.Format("0x{0:X5} is outside the upper range of Unicode code points (0x{1:X5})", start, upperRange))
```

In the future this can be `nameof start`, but that's in preview :)

Reply:
This should be addressed now along with a bunch of other similar cases.